### PR TITLE
Proof-of-concept demonstrating using Neutrino for building

### PR DESCRIPTION
**Pull request by eliperelman**
Do not merge! Just a proof-of-concept for discussion. :)

### CODE REVIEW

**Comment by eliperelman** [comment]
@djmitche good catch; yes that looks right.

**Comment by djmitche** [comment]
I'll give this a shot in deployment.  It will be a good chance to set up a staging environment for secrets.

**Comment by eliperelman** [comment]
Made a few more cleanups, as well as removed `Promise` since the `engines` already requires Node.js v7 which has native Promises.

File: test/mocha.opts [hunk 1]
```diff
@@ -1,4 +0,0 @@
---ui tdd
---timeout 30s
---reporter spec
---require source-map-support/register
```
Comment (eliperelman) [inline test/mocha.opts]: I think this option is still missing from the preset, so I'll look at it.

Comment (eliperelman) [inline test/mocha.opts]: Looks like sourcemaps work fine from tests using the current configuration, so 👍.

**Comment by djmitche** [comment]
Hm, in deploying to Heroku:
```
-----> Installing binaries
       engines.node (package.json):  7.4.0
       engines.npm (package.json):   unspecified (use default)
       
       Downloading and installing node 7.4.0...
       Using default npm version: 4.0.5
       Downloading and installing yarn (0.21.3)...
       Installed yarn 0.21.3
-----> Restoring cache
       Skipping cache restore (disabled by config)
-----> Building dependencies
       Installing node modules (yarn.lock)
       yarn install v0.21.3
       [1/4] Resolving packages...
       [2/4] Fetching packages...
       warning fsevents@1.1.1: The platform "linux" is incompatible with this module.
       info "fsevents@1.1.1" is an optional dependency and failed compatibility check. Excluding it from installation.
       [3/4] Linking dependencies...
       [4/4] Building fresh packages...
       Mon, 08 May 2017 18:16:16 GMT yarn The "child" blocking queue may be stuck. 5 seconds without any activity with 1 worker: /tmp/build_b38f06477172a4ea29e3ec44489cc0e2/taskcluster-taskcluster-secrets-f956d7f/node_modules/libxmljs
       Mon, 08 May 2017 18:16:56 GMT yarn "child" blocking queue finally resolved. Nothing to worry about.
       $ yarn build
       yarn build v0.21.3
       $ neutrino build --use neutrino-preset-taskcluster
       sh: 1: neutrino: not found
       error Command failed with exit code 127.
       error Command failed with exit code 1.
       info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Any ideas what might cause that?

**Comment by eliperelman** [comment]
Yes, Heroku doesn't install devDependencies. :/

**Comment by eliperelman** [comment]
https://devcenter.heroku.com/articles/nodejs-support#devdependencies

**Comment by jonasfj** [comment]
I'm not convinced we want this:
 * `source-map-support` is not recommended for production node
 * without source maps bundling implies garbage stack traces in sentry and logs
 * We don't need to compile code - node have native async/await support

What we do want is a single "preset" containing only:
 * eslint
 * mocha

So we still want to find a way to eliminate boilterplate for these two.

**Comment by djmitche** [comment]
It seems like this has evolved into a larger "is this a good idea" discussion, which will be better-suited to an RFC.  So, closing this for now.

**Pull request closed without merging**